src/main/java/de/tum/in/www1/exerciseapp/web/rest/CourseResource.java

```diff
@@ -150,20 +226,85 @@ public CourseResource(CourseService courseService, AuthorizationCheckService aut
      *
      * @param courseId the Id of the course
      * @return collection of Results where the sum of the best result per exercise, for each student in a course is cointained:
-     *  ResultId refers in this case to the studentId, the score still needs to be divided by the amount of exercises (done in the webapp)
+     * ResultId refers in this case to the studentId, the score still needs to be divided by the amount of exercises (done in the webapp)
      */
     @GetMapping("/courses/{courseId}/getAllCourseScoresOfCourseUsers")
     @PreAuthorize("hasAnyRole('TA', 'INSTRUCTOR', 'ADMIN')")
     @Timed
-    public ResponseEntity<Collection<Result>> getAllSummedScoresOfCourseUsers(@PathVariable("courseId") Long courseId){
+    public ResponseEntity<Collection<Result>> getAllSummedScoresOfCourseUsers(@PathVariable("courseId") Long courseId) {
         log.debug("REST request to get courseScores from course : {}", courseId);
         Course course = courseService.findOne(courseId);
-        if (!authCheckService.isTeachingAssistantInCourse(course) &&
-             !authCheckService.isInstructorInCourse(course) &&
-             !authCheckService.isAdmin()) {
-            return ResponseEntity.status(HttpStatus.FORBIDDEN).build();
-        }
+        if (!userHasPermission(course)) return ResponseEntity.status(HttpStatus.FORBIDDEN).build();
         return ResponseEntity.ok(courseService.getAllOverallScoresOfCourse(courseId));
     }
 
+    private ObjectNode exerciseToJsonWithParticipation(Exercise exercise, List<Participation> participations) {
+        // get user's participation for the exercise
+        Participation participation = null;
+        if (exercise instanceof QuizExercise) {
+            for (Participation participation1 : participations) {
+                if (participation1.getExercise().equals(exercise)) {
+                    participation = participation1;
+                    break;
+                }
+            }
+        } else {
+            for (Participation participation1 : participations) {
+                if (participation1.getExercise().equals(exercise)) {
+                    if (participation1.getInitializationState() == ParticipationState.INITIALIZED) {
```

Comment:
please add comments why we filter for initialized ones and then break and why we also search for inactive ones. Otherwise the logic is not clear

Reply:
Also could you extract another method for finding the correct participation and put this possibly into the Exercise class (respectively subclasses)?